instructions/steps.md

```diff
@@ -493,19 +528,21 @@ before the above `assert`.
 ## Step 10
 
 Ok, so what do we do if there are not enough `:` in `identifier`?
-Dafny does not have a ability to `throw`.
+Dafny does not have a ability to `throw` an exception.
 The return type in Dafny is a contract or postcondition.
-That means that we MUST return an `AwsKmsArn`.
+That means that with the current function signature,
+we MUST return an `AwsKmsArn` no matter what.
 
 What we need is a way to express failure.
-Dafny has a way to do this,
-you can read about [failure compatible types here](https://dafny.org/dafny/DafnyRef/DafnyRef#sec-update-failure)
+Dafny has a way to do this.
+You can read about [failure compatible types here](https://dafny.org/dafny/DafnyRef/DafnyRef#sec-update-failure)
 if you like.
-But we will go over everything you need here.
+But we will go over everything you need.
 
 First we need a type that can express
 the difference between `Success` and `Failure`.
-In the `Wrappers` the `Result` type does exactly this.
+In the `Wrappers` module defined in the `include.dfy` file,
```

Comment:
Hum, nit, it is not defined there, it is only included there

Reply:
Touché :)